File: exercises/c/container_with_most_water.c
#include <stdio.h>

int maxArea(int* height, int heightSize) {
    int l = 0, r = heightSize - 1;
    int ans = 0;
    while (l < r) {
        int area = (height[l] < height[r] ? height[l] : height[r]) * (r - l);
        if (area > ans) {
            ans = area;
        }
        if (height[l] <= height[r]) {
            ++l;
        } else {
            --r;
        }
    }
    return ans;
}

File: exercises/javascript/container_with_most_water.js
function maxArea(height) {
    let l = 0, r = height.length - 1;
    let ans = 0;
    while (l < r) {
        let area = Math.min(height[l], height[r]) * (r - l);
        ans = Math.max(ans, area);
        if (height[l] <= height[r]) {
            l++;
        } else {
            r--;
        }
    }
    return ans;
}
